llvm/include/llvm/IR/IntrinsicsDirectX.td

```diff
@@ -19,4 +19,9 @@ def int_dx_flattened_thread_id_in_group : Intrinsic<[llvm_i32_ty], [], [IntrNoMe
 
 def int_dx_create_handle : ClangBuiltin<"__builtin_hlsl_create_handle">,
     Intrinsic<[ llvm_ptr_ty ], [llvm_i8_ty], [IntrWillReturn]>;
-}
+
+def int_dx_dot : 
+    Intrinsic<[LLVMVectorElementType<0>], 
+    [llvm_anyvector_ty, LLVMScalarOrSameVectorWidth<0, LLVMVectorElementType<0>>],
```

Comment:
Does LLVMMatchType<0> good enough here to replace LLVMScalarOrSameVectorWidth<0, LLVMVectorElementType<0>>?

Reply:
Thats a good question. What I was afraid of is that `LLVMMatchType<0>`  would do its match on the type enum like FixedVector since arg0 is `llvm_anyvector_ty` or if it would also consider match on vector width and element type so I thought to be more explicit. 

for example what I don't want is 

float2 and int2 to match.

Reply:
@python3kgae  Are you ok with leaving this as is?